sample-database-template/modules/create-sk-template.psm1

```diff
@@ -0,0 +1,138 @@
+        noExtensions            = $noExtensions
+        noValidation            = if ($PSBoundParameters.ContainsKey('noValidation')) { $noValidation } else { $true }
+        engine                  = $engine
+        createByRestoringBackup = $createByRestoringBackup
+    }
+
+    $config = (Get-SKConfiguration $paramConfig)
+    Write-FlatHashtable $config
+
+    if ([string]::IsNullOrWhiteSpace($config.createByRestoringBackup)) { $config.createByRestoringBackup = (Get-PopulatedTemplateBackupPathFromSettings $config.appSettings) }
+
+    $script:result = @()
+
+    $elapsed = Use-StopWatch {
+        try {
+            $script:result += Invoke-Task 'Invoke-SetTestHarnessConfig' { Invoke-SetTestHarnessConfig $config }
+            $script:result += Invoke-Task 'Remove-Plugins' { Remove-Plugins $config.appSettings }
+            $script:result += Invoke-Task 'Get-Plugins' { Get-Plugins $config.appSettings }
+            $script:result += Invoke-Task 'Invoke-SampleXmlValidation' { Invoke-SampleXmlValidation $config }
+            $script:result += Invoke-Task 'New-TempDirectory' { New-TempDirectory $config }
+            $script:result += Invoke-Task 'Copy-BootstrapInterchangeFiles' { Copy-BootstrapInterchangeFiles $config }
+            $script:result += Invoke-Task 'Copy-SampleInterchangeFiles' { Copy-SampleInterchangeFiles $config }
+            $script:result += Invoke-Task 'Copy-SchemaFiles' { Copy-SchemaFiles $config }
+            $script:result += Invoke-Task 'Add-RandomKeySecret' { Add-RandomKeySecret $config }
+            $script:result += Invoke-Task 'Invoke-RestoreLoadToolsPackages' { Invoke-RestoreLoadToolsPackages $config }
+            $script:result += Invoke-Task 'Invoke-BuildLoadTools' { Invoke-BuildLoadTools $config }
+            $script:result += Invoke-Task 'New-DatabaseTemplate' { New-DatabaseTemplate $config }
+            $script:result += Invoke-Task 'Assert-DisallowedSchemas' { Assert-DisallowedSchemas $config }
+            $script:result += Invoke-Task 'Invoke-StartTestHarness' { Invoke-StartTestHarness $config }
+            $script:result += Invoke-Task 'Invoke-LoadBootstrapData' { Invoke-LoadBootstrapData $config }
+            $script:result += Invoke-Task 'Invoke-LoadSampleData' { Invoke-LoadSampleData $config }
+            $script:result += Invoke-Task 'Stop-TestHarness' { Stop-TestHarness $config }
+            $script:result += Invoke-Task 'Backup-DatabaseTemplate' { Backup-DatabaseTemplate $config }
+            $script:result += Invoke-Task 'New-DatabaseTemplateNuspec' { New-DatabaseTemplateNuspec $config }
+        }
+        catch {
+            Stop-TestHarness
+            throw $_
+        }
+    }
+
+    Test-Error
+
+    $script:result += New-TaskResult -name '-' -duration '-'
+    $script:result += New-TaskResult -name $MyInvocation.MyCommand.Name -duration $elapsed.format
+
+    return $script:result | Format-Table
+}
+
+Export-ModuleMember -function * -Alias *
```

Comment:
please export the specific functions you are going to use

Reply:
@jamessmckay Export-ModuleMember -Function * -Alias * will export all the functions and aliases defined in the script module.

Reply:
updated